### ISSUE TITLE
Before-hook and method-update, altering the doc

### ISSUE THREAD

**Issue by izaa** [Meteor-Community-Packages/meteor-autoform#1487]
Hi,

I'm using aldeed:simple-schema 1.5.3 and aldeed:autoform 5.8.1 and Meteor 1.3.3

I hava a form but before submitting I want to modify the document. I want to add items to array.

What is strange is what documentation says

```
var hooksObject = {
  before: {
    // Replace `formType` with the form `type` attribute to which this hook applies
    formType: function(doc) {
      // Potentially alter the doc
      doc.foo = 'bar';

      // Then return it or pass it to this.result()
      //return doc; (synchronous)
      //return false; (synchronous, cancel)
      //this.result(doc); (asynchronous)
      //this.result(false); (asynchronous, cancel)
    }
  },
```

My code starts like this

```
before: {
        'method-update': function (doc) {
//
}
}
```

I cannot update doc directly using `doc.feature = 1`, but i can update my doc by `$doc.$set.feature = 1`. But now, if my SimpleSchema collection has _more_ properties than my form has, `console.log(doc.$set)` displays ALL the fields that are in Schema, but the Chrome's debugger can only see fields that are in form.

Which means I cannot see if property has any value `if (doc.$set.myprop === "something")` because myprop is not in form.

What i can do is 

`if (this.template.data.doc.myprop === "something") { }`

TL;DR:
- why console.log and debugger shows differen properties in doc
- can JS have hidden object properies?
- should documentation be updated so we need to use $set-modifier ?
- until latest update, doc.$set saw all the fields even they weren't in form

P.s. feel free to remove this if this is not relevant. I was suprised how i lost some fields from doc.$set,
